pandas/_config/__init__.py

```diff
@@ -33,3 +33,8 @@
 def using_string_dtype() -> bool:
     _mode_options = _global_config["future"]
     return _mode_options["infer_string"]
+
+
+def using_pdep16_nan_behavior() -> bool:
+    _mode_options = _global_config["mode"]
```

Comment:
As this is a POC, feel free to ignore for now, but if we move forward with this, I would prefer to use `future` rather than `mode`, unless we expect to keep the two "modes" indefinitely and allow users to switch between them. Also, we would probably want something different from "pdep16_nan_behavior". Just a comment for future discussion. no need to change now.